switchtec_dma.c

```diff
@@ -611,14 +613,17 @@ static void switchtec_dma_process_desc(struct switchtec_dma_chan *swdma_chan)
 		spin_lock_bh(&swdma_chan->complete_lock);
 
 		ce = switchtec_dma_get_ce(swdma_chan, swdma_chan->cq_tail);
-		if (le16_to_cpu(ce->phase_tag) == swdma_chan->phase_tag) {
+		phase_tag = smp_load_acquire(&ce->phase_tag);
+		if (le16_to_cpu(phase_tag) == swdma_chan->phase_tag) {
 			spin_unlock_bh(&swdma_chan->complete_lock);
 			break;
 		}
 
 		cid = le16_to_cpu(ce->cid);
 		se_idx = cid & (SWITCHTEC_DMA_SQ_SIZE - 1);
 		desc = switchtec_dma_get_desc(swdma_chan, se_idx);
+
+		tail = swdma_chan->tail;
 		cur_desc = switchtec_dma_get_desc(swdma_chan, swdma_chan->tail);
```

Comment:
I was about to suggest using `tail` in this `switchtec_dma_get_desc()` call, but now I see that `cur_desc` is assigned but never actually used, so the whole line can probably be axed.

Reply:
Nice catch. Will remove it.